### PR TITLE
csi: add nodeExpandSecret KEP details

### PR DESCRIPTION
**Pull request by humblec**
Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

Ref # 

KEP: https://github.com/kubernetes/enhancements/pull/3173/
Implementation: https://github.com/kubernetes/kubernetes/pull/105963
Blog: https://github.com/kubernetes/website/pull/32450

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/assign @sftim

**Comment by tengqm** [comment]
/retitle [WIP] csi: add nodeExpandSecret KEP details

**Comment by mehabhalodiya** [comment]
enhancement ref: https://github.com/kubernetes/enhancements/issues/3107

**Review by sftim — COMMENTED** [review]
Ideally, the text that's introduced with:

> The following fields are available to storage administrators to configure a CSI persistent volume:

would move into the reference section. However, we can refactor that after the release. 

---

Here's some suggested wording.

File: content/en/docs/concepts/storage/volumes.md [hunk 1]
```diff
@@ -1247,6 +1247,14 @@ persistent volume:
   `NodePublishVolume` call. This field is optional, and may be empty if no
   secret is required. If the secret object contains more than one secret, all
   secrets are passed.
+* `nodeExpandSecretRef`: A reference to the secret object containing
+  sensitive information to pass to the CSI driver to complete the CSI
+  `NodeExpandVolume` call. This field is optional, and may be empty if no
+  secret is required. If the secret object contains more than one secret, all
```
Comment (divya-mohan0209) [inline content/en/docs/concepts/storage/volumes.md:1253]:
```suggestion
  secret is required. If the object contains more than one secret, all
```

**Review by divya-mohan0209 — CHANGES_REQUESTED** [review]

**Comment by mehabhalodiya** [comment]
The v1.24 milestone no longer tracks the KEP that this PR supports. Thanks for your effort on this documentation; I recommend saving this work for a future release.

/close

If you feel that this PR has been closed in error, please feel free to reopen it against another branch.
Thank you!

**Pull request closed without merging**